packages/ebsi-authorization-client/src/agent/EBSIAuthorizationClient.ts

```diff
@@ -0,0 +1,363 @@
+          Accept: 'application/json',
+        }),
+      })
+    ).json()
+  }
+
+  private async getAccessToken(args: GetAccessTokenArgs): Promise<GetAccessTokenResponse> {
+    const { grant_type = 'vp_token', scope, vp_token, presentation_submission } = args
+    return await (
+      await fetch(`${this.configuration.token_endpoint}`, {
+        method: 'POST',
+        headers: new Headers({
+          ContentType: 'application/x-www-form-urlencoded',
+          Accept: 'application/json',
+        }),
+        body: new URLSearchParams({
+          grant_type,
+          scope,
+          vp_token,
+          presentation_submission: JSON.stringify(presentation_submission),
+        }),
+      })
+    ).json()
+  }
+
+  private async initiateSIOPDidAuthRequest(args: InitiateSIOPDidAuthRequestArgs): Promise<InitiateSIOPDidAuthRequestResponse> {
+    return await (
+      await fetch(`${this.configuration.issuer}/authentication-requests`, {
+        method: 'POST',
+        headers: new Headers({
+          ContentType: 'application/json',
+          Accept: 'application/json',
+        }),
+        body: JSON.stringify(args),
+      })
+    ).json()
+  }
+
+  private async createSIOPSession(args: CreateSIOPSessionArgs): Promise<CreateSIOPSessionResponse> {
+    const formData = new FormData()
+    Object.entries(args).forEach((entry) => formData.append(entry[0], entry[1]))
+    return await (
+      await fetch(`${this.configuration.issuer}/siop-sessions`, {
+        method: 'POST',
+        headers: new Headers({
+          ContentType: 'application/x-www-form-urlencoded',
+          Accept: 'application/json',
+        }),
+        body: formData,
+      })
+    ).json()
+  }
+
+  private async createOAuth2Session(args: CreateOAuth2SessionArgs): Promise<CreateOAuth2SessionResponse> {
+    return await (
+      await fetch(`${this.configuration.issuer}/oauth2-session`, {
+        method: 'POST',
+        headers: new Headers({
+          ContentType: 'application/json',
+          Accept: 'application/json',
+        }),
+        body: JSON.stringify(args),
+      })
+    ).json()
+  }
+}
+
+export async function createVPJwt(
+  args: {
+    keyPair: { publicKeyHex: string; privateKeyHex: string }
+    did: string
+    vc: string
+    audience: string
+    domain: string
+    nbf?: number
+    exp?: number
+    nonce?: number
+  },
+  context: IRequiredContext,
+): Promise<string> {
+  const { keyPair, did, vc, audience, nbf, exp, nonce } = args
+
+  if (!keyPair || !keyPair.privateKeyHex || !keyPair.publicKeyHex) {
+    throw new Error(`Private key is required`)
+  }
+
+  if (!args.vc) {
+    throw new Error('Verifiable Credential not defined')
+  }
+
+  const privateKeyJWT = await exportJWK(u8a.fromString(keyPair.privateKeyHex, 'hex'))
+  const publicKeyJWT = await exportJWK(u8a.fromString(keyPair.publicKeyHex, 'hex'))
+
+  let verifiableCredential: string[]
+
+  if (vc === 'empty') {
+    verifiableCredential = []
```

Comment:
How would no VCs result in a valid VP where the definitions already made clear it expects 1 VC?

Reply:
Something inherited from the EBSI code base. Removed